From 4c0ebc314931b53a150c16d294f5dda7ab834c45 Mon Sep 17 00:00:00 2001
From: "Ethan D. Twardy" <ethan.twardy@gmail.com>
Date: Wed, 11 Oct 2023 21:21:36 -0500
Subject: [PATCH] wayland: Setup sway desktop for work virtual machines

 * Disable hardware cursors
 * Don't configure the apple-dcp driver
 * Use agreety instead of tuigreet

diff --git a/Makefile b/Makefile
index c30bd2d..2f28f42 100644
--- a/Makefile
+++ b/Makefile
@@ -23,5 +23,3 @@ install:
 	install -Dm644 wayland/50-systemd-user.conf -t $(DESTDIR)/etc/sway/config.d
 	: # SSH Agent
 	install -Dm644 ssh-agent.service -t $(DESTDIR)$(systemd_userdir)
-	: # Enable "the notch"
-	install -Dm644 wayland/apple-dcp.conf -t $(DESTDIR)/lib/modprobe.d
diff --git a/wayland/apple-dcp.conf b/wayland/apple-dcp.conf
deleted file mode 100644
index 74e4e9b..0000000
--- a/wayland/apple-dcp.conf
+++ /dev/null
@@ -1 +0,0 @@
-options apple_dcp show_notch=Y
diff --git a/wayland/cobblede.sh b/wayland/cobblede.sh
index f8a01bb..482f96e 100644
--- a/wayland/cobblede.sh
+++ b/wayland/cobblede.sh
@@ -5,3 +5,4 @@ export SDL_VIDEODRIVER=wayland
 export _JAVA_AWT_WM_NONREPARENTING=1
 export XDG_SESSION_DESKTOP=sway
 export XDG_CURRENT_DESKTOP=sway
+export WLR_NO_HARDWARE_CURSORS=1
diff --git a/wayland/greetd.toml b/wayland/greetd.toml
index 9fc7a79..af083a2 100644
--- a/wayland/greetd.toml
+++ b/wayland/greetd.toml
@@ -8,7 +8,7 @@ vt = 1
 
 # `agreety` is the bundled agetty/login-lookalike. You can replace `/bin/sh`
 # with whatever you want started, such as `sway`.
-command = "tuigreet --cmd 'systemd-cat --identifier=wayland /usr/bin/sway'"
+command = "agreety --cmd 'systemd-cat --identifier=wayland /usr/bin/sway'"
 
 # The user to run the command as. The privileges this user must have depends
 # on the greeter. A graphical greeter may for example require the user to be
-- 
2.42.0

